modules/loader/src/main/scala/com/snowplowanalytics/snowplow/rdbloader/redshift/Main.scala

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) 2012-2021 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.snowplow.rdbloader.redshift
+
+import cats.Applicative
+import cats.data.Validated._
+import cats.effect.{Concurrent, ExitCode, IO, IOApp, Timer}
+import cats.implicits._
+
+import com.snowplowanalytics.snowplow.rdbloader.common.Common
+import com.snowplowanalytics.snowplow.rdbloader.common.TableDefinitions.AtomicDefaultColumns
+import com.snowplowanalytics.snowplow.rdbloader.core.config.CliConfig
+import com.snowplowanalytics.snowplow.rdbloader.core.LoaderError
+import com.snowplowanalytics.snowplow.rdbloader.redshift.db.Migration
+import com.snowplowanalytics.snowplow.rdbloader.redshift.db.Manifest
+import com.snowplowanalytics.snowplow.rdbloader.core.discovery.DataDiscovery
+import com.snowplowanalytics.snowplow.rdbloader.core.algebras._
+import com.snowplowanalytics.snowplow.rdbloader.redshift.loading.Load
+import com.snowplowanalytics.snowplow.rdbloader.redshift.db.Statement.{
+  CreateAlertingTempTable,
+  DropAlertingTempTable,
+  FoldersCopy,
+  FoldersMinusManifest
+}
+import com.snowplowanalytics.snowplow.rdbloader.generated.BuildInfo
+import fs2.Stream
+import org.typelevel.log4cats.slf4j.Slf4jLogger
+
+import com.snowplowanalytics.snowplow.rdbloader.core.config.StorageTarget.Redshift
+import com.snowplowanalytics.snowplow.rdbloader.core.algebras.FolderMonitoring.TempTableOps
+
+object Main extends IOApp {
+
+  implicit private val LoggerName =
+    Logging.LoggerName(getClass.getSimpleName.stripSuffix("$"))
+
+  def run(argv: List[String]): IO[ExitCode] =
```

Comment:
Given that all previous comments were addressed. And all dependcies and turned into algebras, this could be moved into the object Runner in core

object Runner {
   def run(argv: List[String])(implicit val dec: Decoder[StorageTarget], implicit val logger: Logger[F])[F[_]: All of the algebras) : F[ExitCode] = ???
} 

Leaving this Main as a collector of interpreters' import, logger setup and import of Decoder. Would be nice right?